.circleci/config.yml

```diff
@@ -0,0 +1,136 @@
+    - run:
+        name: Build packages
+        command: |
+          source ~/.circlerc
+          docker-compose run \
+            -e PKG_VERSION=${PKG_VERSION} \
+            -e PKG_RELEASE=${PKG_RELEASE} \
+            ${DISTRO} build
+    # FIXME: These test containers are 1.6dev, and are out of date
+    # Once containers are updated, the tests need to look for ERROR logs, e.g. due to /stream not reachable
+    - run: 
+        name: Pull & launch Docker st2box images
+        command: |
+          docker-compose -f ~/st2box/docker-compose.yaml pull
+          docker-compose -f ~/st2box/docker-compose.yaml up -d
+    # sleep 10 is needed to allow the st2box test containers time to start up.
+    # Otherwise st2api is not ready when the tests run.
+    - run:
+        name: Run tests
+        command: |
+          source ~/.circlerc
+          sleep 10
+          export ST2_HOSTNAME=$(ifconfig docker0 | grep 'inet addr' | awk -F: '{print $2}' | awk '{print $1}')
+          docker-compose run \
+          -e ST2_HOSTNAME=${ST2_HOSTNAME} \
+          -e ST2_USERNAME=${ST2_USERNAME} \
+          -e ST2_PASSWORD=${ST2_PASSWORD} \
+          -e SLACK_TOKEN=${SLACK_TOKEN} \
+          ${DISTRO}-test test
+    - run:
+        name: Collect logs
+        command: |
+          for name in $(docker ps -a --format "{{.Names}}"); do docker logs ${name} > /tmp/st2chatops/log/${name}.log 2>&1; done
+          mkdir -p ~/packages/${DISTRO}
+          cp -r /tmp/st2chatops/*  ~/packages/${DISTRO}
+    - store_artifacts:
+        path: ~/packages
+        destination: packages
+    - persist_to_workspace:
+        root: ~/
+        paths:
+          - st2-packages
+          - packages
+
+  # Deploy produced deb/rpm packages to PackageCloud staging
+  deploy:
+    docker:
+      # The primary container is an instance of the first list image listed. Your build commands run in this container.
+      - image: circleci/ruby:2.4
+    working_directory: ~/packages
+    environment:
+      DISTROS: "trusty xenial el6 el7"
+    steps:
+      - attach_workspace:
+          at: ~
+      - run:
+          name: List workspace files
+          command: find . | sed 's|[^/]*/|  |g'
+      - run:
+          name: Install dependencies
+          command: |
+            set -x
+            sudo apt-get -y install parallel
+            gem install package_cloud
+      - run:
+          name: Deploy deb/rpm packages to PackageCloud
+          command: |
+            "parallel -v -j0 --line-buffer  ~/st2-packages/.circle/packagecloud.sh deploy {} ~/packages/{} ::: ${DISTROS}"
+            ~/st2-packages/.circle/save_payload.py ~/packages
+
+workflows:
+  version: 2
+  package-test-and-deploy:
+    jobs:
+      - build
+      - deploy:
+          requires:
+            - build
+          filters:
+            branches:
+              only:
+                - master
+                - /v[0-9]+\.[0-9]+/
+                - feature/circleci
```

Comment:
@LindsayHill You can temporary add `circle2.0` branch here to test the deployment step so there won't be surprises during the upcoming release.